### PR TITLE
fix: add context to prevent notion from creating unwanted pages

### PR DESCRIPTION
**Pull request by njhale**
Add context to the Notion tools to confirm with the user before creating new pages. This will prevent pages from being created when the user is actually attempting to update the content of an existing page.

This change also bumps node-gptscript to include the new dataset changes.

Addresses https://github.com/otto8-ai/otto8/issues/251

### CODE REVIEW

**Comment by njhale** [comment]
Running into some dataset issues. Going to see if they're resolved with https://github.com/gptscript-ai/node-gptscript/pull/102

File: notion/package.json
Comment (drpebcak) [inline notion/package.json:10]: I'd rather just wait until grant's changes are merged rather than using his fork in the primary tool repo

**Review by drpebcak — CHANGES_REQUESTED** [review]

File: notion/package.json
Comment (njhale) [reply notion/package.json]: OK -- the dataset changes have merged and I updated this dependency.

**Review by thedadams — APPROVED** [review]

**Review by g-linville — APPROVED** [review]
This LGTM, but I would prefer for it to merge after my PR to this repo, which will conflict with this one but updates all tools in a standard way.

**Pull request merged**
